tests/testthat/test-shinytest2-tm_front_page.R

```diff
@@ -0,0 +1,48 @@
+test_that("e2e: tm_front_page initializes without errors", {
+  skip_if_too_deep(5)
+  require(shinytest2)
+
+  data <- simple_cdisc_data()
+  data <- within(data, {
+    attr(ADSL, "metadata") <- list("Author" = "NEST team", "data_source" = "synthetic data")
+  })
+
+  app <- TealAppDriver$new(
+    data = data,
+    modules = teal::modules(
+      tm_front_page(
+        header_text = c(
+          "Important information" = "It can go here.",
+          "Other information" = "Can go here."
+        ),
+        tables = list("MTCARS" = head(mtcars, 5), "IRIS" = head(iris, 5)),
+        additional_tags = HTML("Additional HTML or shiny tags go here <br>"),
+        footnotes = c("X" = "is the first footnote", "Y is the second footnote"),
+        show_metadata = TRUE
+      )
+    ),
+    timeout = 3000
+  )
+
+  app$expect_no_shiny_error()
+
+  # header text
+  app$expect_html(selector = "#front_page_headers")
+
+  # tables
+  app$expect_screenshot(selector = app$active_module_element("table_1"))
```

Comment:
I do not like when we perform the snapshot tests. They are very fragile. Let's try to find some other way to test this.

Reply:
I see that we moved to snapshot tests after using the hash. They are better to debug compared to hash. Both are fragile. I've already seen TLG catalog snapshot updates after the calculation change being made.
I think we can use them in cases where we're certain that the output will not change, what we have here seems good to me.
Do we need to add the snapshots for all three platforms in the PR? Now it only contains windows and R 4.3
Please also consider using a [seed](https://rstudio.github.io/shinytest2/reference/AppDriver.html#method-new-) during the `TealAppDriver` init when we use snapshots.

Reply:
do you prefer if we just do `expect_html `as they are HTML elements.

Reply:
maybe `$expect_html` and verify `$get_text()`

Reply:
Screenshots not only area prone to R version and platform, but also to system fonts : ) 

<img width="322" alt="image" src="https://github.com/insightsengineering/teal.modules.general/assets/133694481/0f9f77ae-eed3-4523-a626-dca936f8133d">

Reply:
@vedhav, is snapshot and screenshot something else? Kartik had screenshots (images) in here, and snaphosts are stored JSONs?

Reply:
I was referring snapshot to as any kind of "store the output -> check it later" which includes images, json or any other file format.
They are great for regression testing, where we need to make sure that the UI does not change, shinytest2 is primarily meant for these tests. At least we are testing just a particular selector area for screenshots so we're okay. But, I still don't like them.

Reply:
There is no random-ness on this module so I think it plays well here.

It tracks the specific tables instead of the whole app or figures.

Do you have an alternative in mind?

Reply:
I can update on ubuntu